R/grid.R

```diff
@@ -139,7 +139,7 @@ st_viewport = function(x, ..., bbox = st_bbox(x), asp) {
 
 #' @export
 st_as_grob.sfc_POINT <- function(x, pch = 1, size = unit(1, "char"), default.units = "native", name = NULL, gp = gpar(), vp = NULL, ...) {
-	x <- matrix(unlist(x), nrow = 2)
+	x <- matrix(unlist(x, use.names = FALSE), ncol = length(x))
```

Comment:
I think that `ncol` should be `nrow`, here.

Reply:
I don't think so. The current code uses `nrow` because it specifies the dimension (2D). On the other hand, this PR uses `length(x)`, which is not the dimension of data, but the size of the data. So, I think this should be `ncol`. Does this make sense...?

``` r
library(sf)
#> Linking to GEOS 3.7.2, GDAL 2.4.2, PROJ 6.1.1

pt <- list(st_point(c(1, 1, 1)), st_point(c(2, 2, 2)))
sfc <- st_sfc(pt)

unlist(sfc)
#> [1] 1 1 1 2 2 2

m <- matrix(unlist(sfc), nrow = length(sfc))
m
#>      [,1] [,2] [,3]
#> [1,]    1    1    2
#> [2,]    1    2    2
cat("x:", m[1, ], ", y:", m[2, ])
#> x: 1 1 2 , y: 1 2 2

m <- matrix(unlist(sfc), ncol = length(sfc))
m
#>      [,1] [,2]
#> [1,]    1    2
#> [2,]    1    2
#> [3,]    1    2
cat("x:", m[1, ], ", y:", m[2, ])
#> x: 1 2 , y: 1 2
```

<sup>Created on 2019-12-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Ah, I get it - thanks!